codex-rs/core/src/shell.rs

```diff
@@ -116,13 +116,45 @@ fn strip_bash_lc(command: &Vec<String>) -> Option<String> {
     }
 }
 
+fn get_zsh_shell() -> Option<ZshShell> {
```

Comment:
Should we do:

```suggestion
#[cfg(not(unix))]
fn get_zsh_shell() -> Option<ZshShell> {
    None
}

#[cfg(unix)]
fn get_zsh_shell() -> Option<ZshShell> {
```

And then can we remove most of the `#[cfg(target_os = "macos")]` stuff or will we get a dead code warning on Windows arguing that `ZshShell` is never constructed?